### PR TITLE
Should not be displayed without []

### PR DESCRIPTION
**Pull request by 9kgsofrice**
This is cleaner now and should work

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/Building.kt [hunk 1]
```diff
@@ -247,6 +247,12 @@ class Building : NamedStats(), IConstruction {
                         && !construction.containsBuildingOrEquivalent(it.params[0])} )
             return "Should not be displayed"
 
+        if(uniqueObjects.any { it.placeholderText=="Not displayed as an available construction without []"
+                        && ((it.params[0] in civInfo.gameInfo.ruleSet.buildings &&!construction.containsBuildingOrEquivalent(it.params[0]))
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/Building.kt:251]: A. Add a val for `it.params[0]`, call it "filter" or something
B. Remove unnecessary parentheses, if you're using Android Studio it'll recommend this itself

Comment (yairm210) [reply core/src/com/unciv/models/ruleset/Building.kt:251]: Actually now that I think of it there shouldn't be a reason to have the code for the units and buildings different at all, no?

Comment (9kgsofrice) [reply core/src/com/unciv/models/ruleset/Building.kt]: no, it makes sense that they are the same :)

Comment (9kgsofrice) [reply core/src/com/unciv/models/ruleset/Building.kt]: android studio complains if i remove any ( ), which ones are unnecessary?

Comment (yairm210) [reply core/src/com/unciv/models/ruleset/Building.kt]: Never mind, it's not that important ;)

**Pull request merged**